test/Datum.ts

```diff
@@ -186,6 +186,47 @@ describe('Datum', () => {
     );
   });
 
+  /**
+   * 1. Identity: `A.ap(A.of(a => a), fa) = fa`
+   * 2. Homomorphism: `A.ap(A.of(ab), A.of(a)) = A.of(ab(a))`
+   * 3. Interchange: `A.ap(fab, A.of(a)) = A.ap(A.of(ab => ab(a)), fab)`
+   */
+   it('ap', () => {
+    const f = (n: number) => n * 2;
+
+    assert.deepStrictEqual(D.Apply.ap(D.initial, D.initial), D.initial);
+    assert.deepStrictEqual(D.Apply.ap(D.initial, D.pending), D.initial);
+    assert.deepStrictEqual(D.Apply.ap(D.initial, D.refresh(1)), D.initial);
+    assert.deepStrictEqual(D.Apply.ap(D.initial, D.replete(1)), D.initial);
+
+    assert.deepStrictEqual(D.Apply.ap(D.pending, D.initial), D.pending);
+    assert.deepStrictEqual(D.Apply.ap(D.pending, D.pending), D.pending);
+    assert.deepStrictEqual(D.Apply.ap(D.pending, D.refresh(1)), D.pending);
+    assert.deepStrictEqual(D.Apply.ap(D.pending, D.replete(1)), D.pending);
+
+    assert.deepStrictEqual(D.Apply.ap(D.refresh(f), D.initial), D.initial);
+    assert.deepStrictEqual(D.Apply.ap(D.refresh(f), D.pending), D.pending);
+    assert.deepStrictEqual(
+      D.Apply.ap(D.refresh(f), D.refresh(1)),
+      D.refresh(2)
+    );
+    assert.deepStrictEqual(
+      D.Apply.ap(D.refresh(f), D.replete(1)),
+      D.replete(2)
```

Comment:
Note the mega instance above equals `D.refresh(2)`

https://github.com/nullpub/datum/pull/17/files#diff-c93edea33a3b97edd6aeec7ec663842689e3d415c918fcdd04c98192cae05a0eL174

The new `ap` instance takes the "right-most" refresh state if the "left datum" has a value.

This matches chain. When you think about datums being evaluated sequentially from left-to-right. If it's Valued, the `(a: A) => Datum<B>` function is evaluated, returning the functions returned value as-is.